pointcloud/basic_octree_utils.go

```diff
@@ -93,41 +111,48 @@ func (octree *BasicOctree) checkPointPlacement(p r3.Vector) bool {
 // helperSet is used by Set to recursive move through a basic octree while tracking recursion depth.
 // If the maximum recursion depth is reached before a valid node is found for the point it will return
 // an error.
-func (octree *BasicOctree) helperSet(p r3.Vector, d Data, recursionDepth int) error {
+func (octree *BasicOctree) helperSet(p r3.Vector, d Data, recursionDepth int) (float64, error) {
 	if recursionDepth >= maxRecursionDepth {
-		return errors.New("error max allowable recursion depth reached")
+		return 0, errors.New("error max allowable recursion depth reached")
 	}
 	if (PointAndData{P: p, D: d} == PointAndData{}) {
-		return nil
+		return 0, nil
 	}
 
 	if !octree.checkPointPlacement(p) {
-		return errors.New("error point is outside the bounds of this octree")
+		return 0, errors.New("error point is outside the bounds of this octree")
 	}
 
+	var err error
 	switch octree.node.nodeType {
 	case internalNode:
 		for _, childNode := range octree.node.children {
 			if childNode.checkPointPlacement(p) {
-				err := childNode.helperSet(p, d, recursionDepth+1)
+				mv, err := childNode.helperSet(p, d, recursionDepth+1)
 				if err == nil {
 					// Update metadata
 					octree.meta.Merge(p, d)
 					octree.size++
+					if math.IsNaN(octree.node.maxVal) {
```

Comment:
[nit] to simplify I would do 
```
if !math.IsNan(mv) {
    octree.node.maxVal = mv
}
```
so no need for else clause

Reply:
i don't think this would work.
Suppose at time = 0 we have mv = 10. then octree.node.maxVal = 10
Suppose at time = 1 we have mv = 1. then octree.node.maxVal = 1 which should not happen.

Further:
```
	a := math.NaN()
	fmt.Println(a)
	b := 1.0
	fmt.Println(math.Max(a, b))
```
this prints NaN

Reply:
since we now use int, no need to check for NaN. always just do math.Max()